<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Jacques Serizay">
<title>Hi-C analysis with Bioconductor&lt;br&gt;Package Demo&lt;br&gt;Bioc2023 Conference â€“ visualizing</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
</head>
<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./visualizing.html">Visualizing Hi-C data in R</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Hi-C analysis with Bioconductor<br>Package Demo<br>Bioc2023 Conference</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./slides.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Package presentation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./parsing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Parsing Hi-C data in R</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./visualizing.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Visualizing Hi-C data in R</span></a>
  </div>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li>
<a href="#visualizing-hi-c-data-in-r" id="toc-visualizing-hi-c-data-in-r" class="nav-link active" data-scroll-target="#visualizing-hi-c-data-in-r">Visualizing Hi-C data in R</a>
  <ul class="collapse">
<li>
<a href="#visualizing-hi-c-contact-maps" id="toc-visualizing-hi-c-contact-maps" class="nav-link" data-scroll-target="#visualizing-hi-c-contact-maps">Visualizing Hi-C contact maps</a>
  <ul class="collapse">
<li><a href="#square-matrices" id="toc-square-matrices" class="nav-link" data-scroll-target="#square-matrices">Square matrices</a></li>
  <li><a href="#side-by-side-matrices" id="toc-side-by-side-matrices" class="nav-link" data-scroll-target="#side-by-side-matrices">Side-by-side matrices</a></li>
  <li><a href="#plotting-multiple-chromosomes" id="toc-plotting-multiple-chromosomes" class="nav-link" data-scroll-target="#plotting-multiple-chromosomes">Plotting multiple chromosomes</a></li>
  </ul>
</li>
  <li>
<a href="#tuning-hi-c-contact-maps" id="toc-tuning-hi-c-contact-maps" class="nav-link" data-scroll-target="#tuning-hi-c-contact-maps">Tuning Hi-C contact maps</a>
  <ul class="collapse">
<li><a href="#choosing-scores" id="toc-choosing-scores" class="nav-link" data-scroll-target="#choosing-scores">Choosing scores</a></li>
  <li><a href="#choosing-scale" id="toc-choosing-scale" class="nav-link" data-scroll-target="#choosing-scale">Choosing scale</a></li>
  <li><a href="#choosing-color-map" id="toc-choosing-color-map" class="nav-link" data-scroll-target="#choosing-color-map">Choosing color map</a></li>
  </ul>
</li>
  <li><a href="#read-more" id="toc-read-more" class="nav-link" data-scroll-target="#read-more">Read more</a></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info">Session info</a></li>
  </ul>
</li>
  </ul><div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/js2264/OHCA.Bioc2023/edit/devel/visualizing.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/js2264/OHCA.Bioc2023/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><section id="visualizing-hi-c-data-in-r" class="level1"><h1>Visualizing Hi-C data in R</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Docker image
</div>
</div>
<div class="callout-body-container callout-body">
<p>All the packages used in this demo are available in the following <code>docker</code> image:</p>
<div class="cell" data-layout-align="center" data-hash="visualizing_cache/html/unnamed-chunk-2_56f0c78b5e008b11f93c92fc9db2f467">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>sh</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode sh code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">docker</span> run ghcr.io/js2264/ohca</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Generating the example <code>hic</code> object ðŸ‘‡
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>To demonstrate how to visualize a <code>HiCExperiment</code> contact matrix, we will create an <code>HiCExperiment</code> object from an example <code>.cool</code> file provided in the <code>HiContactsData</code> package.</p>
<div class="cell" data-layout-align="center" data-hash="visualizing_cache/html/unnamed-chunk-3_5264b868d7c1f03df67069ef6fe70693">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/js2264/HiCExperiment">HiCExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/js2264/HiContactsData">HiContactsData</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># ---- This downloads an example `.mcool` file and caches it locally </span></span>
<span><span class="va">coolf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html">HiContactsData</a></span><span class="op">(</span><span class="st">'yeast_wt'</span>, <span class="st">'mcool'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># ---- This creates a connection to the disk-stored `.mcool` file</span></span>
<span><span class="va">cf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HiCExperiment/man/CoolFile-class.html">CoolFile</a></span><span class="op">(</span><span class="va">coolf</span><span class="op">)</span></span>
<span><span class="va">cf</span></span>
<span></span>
<span><span class="co"># ---- This imports contacts from the chromosome `V` at resolution `2000`</span></span>
<span><span class="va">hic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocIO/man/IO.html">import</a></span><span class="op">(</span><span class="va">cf</span>, focus <span class="op">=</span> <span class="st">'V'</span>, resolution <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center" data-hash="visualizing_cache/html/unnamed-chunk-4_b9b6fb3304f0fa90c12f541ff5bcbc1c">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hic</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
<section id="visualizing-hi-c-contact-maps" class="level2"><h2 class="anchored" data-anchor-id="visualizing-hi-c-contact-maps">Visualizing Hi-C contact maps</h2>
<section id="square-matrices" class="level3"><h3 class="anchored" data-anchor-id="square-matrices">Square matrices</h3>
<div class="cell" data-layout-align="center" data-hash="visualizing_cache/html/unnamed-chunk-5_b42426685604c3b0d0a6d61d6ad53ba1">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/js2264/HiContacts">HiContacts</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/HiContacts/man/plotMatrix.html">plotMatrix</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</section><section id="side-by-side-matrices" class="level3"><h3 class="anchored" data-anchor-id="side-by-side-matrices">Side-by-side matrices</h3>
<div class="cell" data-layout-align="center" data-hash="visualizing_cache/html/unnamed-chunk-6_ffb889ad401ccbf54578a2a2d2a930f5">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hic2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocIO/man/IO.html">import</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/HiCExperiment/man/CoolFile-class.html">CoolFile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html">HiContactsData</a></span><span class="op">(</span><span class="st">'yeast_eco1'</span>, <span class="st">'mcool'</span><span class="op">)</span><span class="op">)</span>, </span>
<span>    focus <span class="op">=</span> <span class="st">'V'</span>, </span>
<span>    resolution <span class="op">=</span> <span class="fl">2000</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/HiContacts/man/plotMatrix.html">plotMatrix</a></span><span class="op">(</span><span class="va">hic</span>, compare.to <span class="op">=</span> <span class="va">hic2</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</section><section id="plotting-multiple-chromosomes" class="level3"><h3 class="anchored" data-anchor-id="plotting-multiple-chromosomes">Plotting multiple chromosomes</h3>
<div class="cell" data-layout-align="center" data-hash="visualizing_cache/html/unnamed-chunk-7_811453d8958243a14b4aef4ad3df7de7">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">full_hic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocIO/man/IO.html">import</a></span><span class="op">(</span><span class="va">cf</span>, resolution <span class="op">=</span> <span class="fl">4000</span><span class="op">)</span></span>
<span><span class="va">hic_subset</span> <span class="op">&lt;-</span> <span class="va">full_hic</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"III"</span>, <span class="st">"IV"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/HiContacts/man/plotMatrix.html">plotMatrix</a></span><span class="op">(</span><span class="va">hic_subset</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</section></section><section id="tuning-hi-c-contact-maps" class="level2"><h2 class="anchored" data-anchor-id="tuning-hi-c-contact-maps">Tuning Hi-C contact maps</h2>
<section id="choosing-scores" class="level3"><h3 class="anchored" data-anchor-id="choosing-scores">Choosing scores</h3>
<div class="cell" data-layout-align="center" data-hash="visualizing_cache/html/unnamed-chunk-8_b361acf21bf88854a8e3920a84bbc1bc">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/HiContacts/man/plotMatrix.html">plotMatrix</a></span><span class="op">(</span><span class="va">hic</span>, use.scores <span class="op">=</span> <span class="st">'count'</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</section><section id="choosing-scale" class="level3"><h3 class="anchored" data-anchor-id="choosing-scale">Choosing scale</h3>
<div class="cell" data-layout-align="center" data-hash="visualizing_cache/html/unnamed-chunk-9_dafc539aab9294c0ae03d9b520a26d1a">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/HiContacts/man/plotMatrix.html">plotMatrix</a></span><span class="op">(</span><span class="va">hic</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3.5</span>, <span class="op">-</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</section><section id="choosing-color-map" class="level3"><h3 class="anchored" data-anchor-id="choosing-color-map">Choosing color map</h3>
<div class="cell" data-layout-align="center" data-hash="visualizing_cache/html/unnamed-chunk-10_94df86b6dae3ce77aa63728a3279257a">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># ----- `afmhotr` color map is shipped in the `HiContacts` package</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/HiContacts/man/palettes.html">afmhotrColors</a></span><span class="op">(</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/HiContacts/man/plotMatrix.html">plotMatrix</a></span><span class="op">(</span></span>
<span>    <span class="va">hic</span>, </span>
<span>    use.scores <span class="op">=</span> <span class="st">'balanced'</span>,</span>
<span>    limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">4</span>, <span class="op">-</span><span class="fl">1</span><span class="op">)</span>,</span>
<span>    cmap <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/HiContacts/man/palettes.html">afmhotrColors</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</section></section><section id="read-more" class="level2"><h2 class="anchored" data-anchor-id="read-more">Read more</h2>
<p>Read the following sections from the OHCA book:</p>
<ul>
<li><a href="https://jserizay.com/OHCA/visualization.html">HiCExperiment visualization</a></li>
</ul></section><section id="session-info" class="level2"><h2 class="anchored" data-anchor-id="session-info">Session info</h2>
<div class="cell" data-layout-align="center" data-hash="visualizing_cache/html/unnamed-chunk-11_90313ed7886b0d93bebe66d9d0b5d4f1">
<pre><code>##  â”€ Session info â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
##   setting  value
##   version  R version 4.3.0 (2023-04-21)
##   os       Ubuntu 18.04.6 LTS
##   system   x86_64, linux-gnu
##   ui       X11
##   language (EN)
##   collate  en_US.UTF-8
##   ctype    en_US.UTF-8
##   tz       Europe/Paris
##   date     2023-07-28
##   pandoc   2.13 @ /usr/bin/ (via rmarkdown)
##  
##  â”€ Packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
##   package                * version   date (UTC) lib source
##   abind                    1.4-5     2016-07-21 [1] CRAN (R 4.3.0)
##   AnnotationDbi            1.62.2    2023-07-02 [1] Bioconductor
##   AnnotationHub          * 3.8.0     2023-04-25 [1] Bioconductor
##   beeswarm                 0.4.0     2021-06-01 [1] CRAN (R 4.3.0)
##   Biobase                  2.60.0    2023-04-25 [1] Bioconductor
##   BiocFileCache          * 2.8.0     2023-04-25 [1] Bioconductor
##   BiocGenerics           * 0.46.0    2023-04-25 [1] Bioconductor
##   BiocIO                   1.10.0    2023-04-25 [1] Bioconductor
##   BiocManager              1.30.21.1 2023-07-18 [1] CRAN (R 4.3.0)
##   BiocParallel             1.34.2    2023-05-22 [1] Bioconductor
##   BiocVersion              3.17.1    2022-11-04 [1] Bioconductor
##   Biostrings               2.68.1    2023-05-16 [1] Bioconductor
##   bit                      4.0.5     2022-11-15 [1] CRAN (R 4.3.0)
##   bit64                    4.0.5     2020-08-30 [1] CRAN (R 4.3.0)
##   bitops                   1.0-7     2021-04-24 [1] CRAN (R 4.3.0)
##   blob                     1.2.4     2023-03-17 [1] CRAN (R 4.3.0)
##   cachem                   1.0.8     2023-05-01 [1] CRAN (R 4.3.0)
##   cli                      3.6.1     2023-03-23 [1] CRAN (R 4.3.0)
##   codetools                0.2-19    2023-02-01 [1] CRAN (R 4.3.0)
##   colorspace               2.1-0     2023-01-23 [1] CRAN (R 4.3.0)
##   crayon                   1.5.2     2022-09-29 [1] CRAN (R 4.3.0)
##   curl                     5.0.1     2023-06-07 [1] CRAN (R 4.3.0)
##   DBI                      1.1.3     2022-06-18 [1] CRAN (R 4.3.0)
##   dbplyr                 * 2.3.3     2023-07-07 [1] CRAN (R 4.3.0)
##   DelayedArray             0.26.6    2023-07-02 [1] Bioconductor
##   digest                   0.6.33    2023-07-07 [1] CRAN (R 4.3.0)
##   dplyr                    1.1.2     2023-04-20 [1] CRAN (R 4.3.0)
##   ellipsis                 0.3.2     2021-04-29 [1] CRAN (R 4.3.0)
##   evaluate                 0.21      2023-05-05 [1] CRAN (R 4.3.0)
##   ExperimentHub          * 2.8.1     2023-07-12 [1] Bioconductor
##   fansi                    1.0.4     2023-01-22 [1] CRAN (R 4.3.0)
##   fastmap                  1.1.1     2023-02-24 [1] CRAN (R 4.3.0)
##   filelock                 1.0.2     2018-10-05 [1] CRAN (R 4.3.0)
##   generics                 0.1.3     2022-07-05 [1] CRAN (R 4.3.0)
##   GenomeInfoDb             1.36.1    2023-06-21 [1] Bioconductor
##   GenomeInfoDbData         1.2.10    2023-04-02 [1] Bioconductor
##   GenomicRanges            1.52.0    2023-04-25 [1] Bioconductor
##   ggbeeswarm               0.7.2     2023-04-29 [1] CRAN (R 4.3.0)
##   ggplot2                  3.4.2     2023-04-03 [1] CRAN (R 4.3.0)
##   ggrastr                  1.0.2     2023-06-01 [1] CRAN (R 4.3.0)
##   glue                     1.6.2     2022-02-24 [1] CRAN (R 4.3.0)
##   gtable                   0.3.3     2023-03-21 [1] CRAN (R 4.3.0)
##   HiCExperiment          * 1.1.1     2023-07-27 [1] Github (js2264/HiCExperiment@9e8f7bb)
##   HiContacts             * 1.3.1     2023-07-27 [1] Github (js2264/HiContacts@d3bdfef)
##   HiContactsData         * 1.3.0     2023-05-26 [1] Github (js2264/HiContactsData@adcb51f)
##   hms                      1.1.3     2023-03-21 [1] CRAN (R 4.3.0)
##   htmltools                0.5.5     2023-03-23 [1] CRAN (R 4.3.0)
##   htmlwidgets              1.6.2     2023-03-17 [1] CRAN (R 4.3.0)
##   httpuv                   1.6.11    2023-05-11 [1] CRAN (R 4.3.0)
##   httr                     1.4.6     2023-05-08 [1] CRAN (R 4.3.0)
##   InteractionSet           1.29.1    2023-06-13 [1] Github (LTLA/InteractionSet@62cd50a)
##   interactiveDisplayBase   1.38.0    2023-04-25 [1] Bioconductor
##   IRanges                  2.34.1    2023-06-22 [1] Bioconductor
##   jsonlite                 1.8.7     2023-06-29 [1] CRAN (R 4.3.0)
##   KEGGREST                 1.40.0    2023-04-25 [1] Bioconductor
##   knitr                    1.43      2023-05-25 [1] CRAN (R 4.3.0)
##   later                    1.3.1     2023-05-02 [1] CRAN (R 4.3.0)
##   lattice                  0.21-8    2023-04-05 [1] CRAN (R 4.3.0)
##   lifecycle                1.0.3     2022-10-07 [1] CRAN (R 4.3.0)
##   magrittr                 2.0.3     2022-03-30 [1] CRAN (R 4.3.0)
##   Matrix                   1.6-0     2023-07-08 [1] CRAN (R 4.3.0)
##   MatrixGenerics           1.12.2    2023-06-09 [1] Bioconductor
##   matrixStats              1.0.0     2023-06-02 [1] CRAN (R 4.3.0)
##   memoise                  2.0.1     2021-11-26 [1] CRAN (R 4.3.0)
##   mime                     0.12      2021-09-28 [1] CRAN (R 4.3.0)
##   munsell                  0.5.0     2018-06-12 [1] CRAN (R 4.3.0)
##   pillar                   1.9.0     2023-03-22 [1] CRAN (R 4.3.0)
##   pkgconfig                2.0.3     2019-09-22 [1] CRAN (R 4.3.0)
##   png                      0.1-8     2022-11-29 [1] CRAN (R 4.3.0)
##   promises                 1.2.0.1   2021-02-11 [1] CRAN (R 4.3.0)
##   purrr                    1.0.1     2023-01-10 [1] CRAN (R 4.3.0)
##   R6                       2.5.1     2021-08-19 [1] CRAN (R 4.3.0)
##   rappdirs                 0.3.3     2021-01-31 [1] CRAN (R 4.3.0)
##   Rcpp                     1.0.11    2023-07-06 [1] CRAN (R 4.3.0)
##   RCurl                    1.98-1.12 2023-03-27 [1] CRAN (R 4.3.0)
##   readr                    2.1.4     2023-02-10 [1] CRAN (R 4.3.0)
##   rhdf5                    2.44.0    2023-04-25 [1] Bioconductor
##   rhdf5filters             1.12.1    2023-04-30 [1] Bioconductor
##   Rhdf5lib                 1.22.0    2023-04-25 [1] Bioconductor
##   rlang                    1.1.1     2023-04-28 [1] CRAN (R 4.3.0)
##   rmarkdown                2.23      2023-07-01 [1] CRAN (R 4.3.0)
##   RSpectra                 0.16-1    2022-04-24 [1] CRAN (R 4.3.0)
##   RSQLite                  2.3.1     2023-04-03 [1] CRAN (R 4.3.0)
##   rstudioapi               0.15.0    2023-07-07 [1] CRAN (R 4.3.0)
##   S4Arrays                 1.0.5     2023-07-24 [1] Bioconductor
##   S4Vectors                0.38.1    2023-05-02 [1] Bioconductor
##   scales                   1.2.1     2022-08-20 [1] CRAN (R 4.3.0)
##   sessioninfo              1.2.2     2021-12-06 [1] CRAN (R 4.3.0)
##   shiny                    1.7.4.1   2023-07-06 [1] CRAN (R 4.3.0)
##   strawr                   0.0.91    2023-03-29 [1] CRAN (R 4.3.0)
##   stringi                  1.7.12    2023-01-11 [1] CRAN (R 4.3.0)
##   stringr                  1.5.0     2022-12-02 [1] CRAN (R 4.3.0)
##   SummarizedExperiment     1.30.2    2023-06-06 [1] Bioconductor
##   tibble                   3.2.1     2023-03-20 [1] CRAN (R 4.3.0)
##   tidyr                    1.3.0     2023-01-24 [1] CRAN (R 4.3.0)
##   tidyselect               1.2.0     2022-10-10 [1] CRAN (R 4.3.0)
##   tzdb                     0.4.0     2023-05-12 [1] CRAN (R 4.3.0)
##   utf8                     1.2.3     2023-01-31 [1] CRAN (R 4.3.0)
##   vctrs                    0.6.3     2023-06-14 [1] CRAN (R 4.3.0)
##   vipor                    0.4.5     2017-03-22 [1] CRAN (R 4.3.0)
##   vroom                    1.6.3     2023-04-28 [1] CRAN (R 4.3.0)
##   xfun                     0.39      2023-04-20 [1] CRAN (R 4.3.0)
##   xtable                   1.8-4     2019-04-21 [1] CRAN (R 4.3.0)
##   XVector                  0.40.0    2023-04-25 [1] Bioconductor
##   yaml                     2.3.7     2023-01-23 [1] CRAN (R 4.3.0)
##   zlibbioc                 1.46.0    2023-04-25 [1] Bioconductor
##  
##   [1] /home/rsg/R/x86_64-pc-linux-gnu-library/4.3
##   [2] /usr/local/lib/R-4.3/lib/R/library
##  
##  â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</code></pre>
</div>


</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">OHCA â€“ Package demonstration â€“ Bioc2023 Conference</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">This website was built with <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>


</body></html>